### ISSUE TITLE
Feature request: graceful degradation when network log writer is unavailable

### ISSUE THREAD

**Issue by dtelyukh** [caddyserver/caddy#5520]
If network writer is unavailable, Caddy fails on start:

```
2023/05/04 15:52:32.809	INFO	using provided configuration	{"config_file": "Caddyfile", "config_adapter": ""}
Error: loading initial config: loading new config: setting up custom log 'reverse_proxy': opening log writer using &logging.NetWriter{Address:"localhost:8888", DialTimeout:1000000000, addr:caddy.NetworkAddress{Network:"tcp", Host:"localhost", StartPort:0x22b8, EndPort:0x22b8}}: dial tcp [::1]:8888: connect: connection refused
```

It would be better if it could start without network logger, than to fail.

**Comment by mholt** [comment]
Hm :thinking: 

Starting without logging -- knowingly -- sounds like a bad idea. Why would you want that?

**Comment by dtelyukh** [comment]
If a remote collector  is down, I want to start Caddy anyway. I can see these logs in console. And I want to have working instance of Caddy while I will fix collector. 
I think remote log collector could not be required dependency. In addition, when Caddy is already run and a collector is stopped, Caddy has not any fatal error and works in normal mode.

**Comment by mholt** [comment]
There's a big difference between starting a server knowing that something is broken, and keeping a running server running -- we do the latter at all costs, but the former, still seems like a bad idea to me. But I do understand how a remote dependency is not an ideal blocker for a web server. In v2 we eased up on requiring certificates to succeed before running (granted, if you try to access your site before Caddy can get a cert, you'll get an error) -- so maybe we can do something similar for logs...

**Comment by mholt** [comment]
@dtelyukh I've implemented this in f3e8b9d, and follow-up commit cdce452edc5e9cf9127789a868d01864d3276af5.

I don't feel like it's a good default to ignore log errors since I bet many users rely on it, but we can change this later probably if it turns out to be the preferred default.

Simply add the `soft_start` parameter to your `net` config in the Caddyfile:

```
{
	log default {
		output net foobar
		soft_start
	}
}
```

Or in the JSON, set `soft_start` to `true`.
